<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r setup, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%",
  dpi=300,fig.width=7,
  fig.keep="all"
)
```

# CascadeData <img src="man/figures/logo.png" align="right" width="200"/>

# Experimental Data of Cascade Experiments in Genomics
## Frédéric Bertrand and Myriam Maumy-Bertrand

<https://doi.org/10.32614/CRAN.package.bigalgebra>

<!-- badges: start -->
[![DOI](https://img.shields.io/badge/doi-10.32614/CRAN.package.ModStatR-blue.svg)](https://doi.org/10.32614/CRAN.package.ModStatR)
[![Lifecycle: stable](https://img.shields.io/badge/lifecycle-stable-green.svg)](https://lifecycle.r-lib.org/articles/stages.html)
[![Project Status: Active – The project has reached a stable, usable state and is being actively developed.](https://www.repostatus.org/badges/latest/active.svg)](https://www.repostatus.org/#active)
[![R-CMD-check](https://github.com/fbertran/CascadeData/workflows/R-CMD-check/badge.svg)](https://github.com/fbertran/CascadeData/actions)
[![Codecov test coverage](https://codecov.io/gh/fbertran/CascadeData/branch/master/graph/badge.svg)](https://app.codecov.io/gh/fbertran/CascadeData?branch=master)
[![CRAN status](https://www.r-pkg.org/badges/version/CascadeData)](https://cran.r-project.org/package=CascadeData)
[![CRAN RStudio mirror downloads](https://cranlogs.r-pkg.org/badges/CascadeData)](https://cran.r-project.org/package=CascadeData)
[![GitHub Repo stars](https://img.shields.io/github/stars/fbertran/CascadeData?style=social)](https://github.com/fbertran/CascadeData)
<!-- badges: end -->


The goal of CascadeData is to provide the experimental data [GSE39411](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411) in a ready to use format. Vallat L, Kemper CA, Jung N, Maumy-Bertrand M, Bertrand F, ..., Bahram S, (2013), "Reverse-engineering the genetic circuitry of a cancer cell with predicted intervention in chronic lymphocytic leukemia". *Proc Natl Acad Sci USA*, **110**(2):459-64, <https://dx.doi.org/10.1073/pnas.1211130110>.

These are featured as examples by packages such as the Cascade one, a modeling tool allowing gene selection, reverse engineering, and prediction in cascade networks. (Jung, N., Bertrand, F., Bahram, S., Vallat, L., and Maumy-Bertrand, M., 2014, <https://dx.doi.org/10.1093/bioinformatics/btt705>).

This website and these examples were created by F. Bertrand and M. Maumy-Bertrand.


## Installation

You can install the released version of CascadeData from [CRAN](https://CRAN.R-project.org) with:

```{r, eval = FALSE}
install.packages("CascadeData")
```

You can install the development version of CascadeData from [github](https://github.com) with:

```{r, eval = FALSE}
devtools::install_github("fbertran/CascadeData")
```
## Examples

### Probesets
Two data frames with 54613 probesets with repeated measurements on 6 independent subjects at 4 time points.

```{r exampleS}
data(micro_S)
str(micro_S)
```

```{r exampleUS}
data(micro_US)
str(micro_US)
```

Some preliminary between group comparison. First create grouping factor.
```{r grouping factor}
groupf=factor(c(rep("S",ncol(micro_S)),rep("US",ncol(micro_US))))
```

Then, create the 2 most discriminative components (probeset linear combinaison, i.e. scores) featuring 100 probesets each using sparse partial least squares discrimant analysis from the [mixOmics](https://www.bioconductor.org/packages/release/bioc/html/mixOmics.html) package, [doi:10.18129/B9.bioc.mixOmics](https://doi.org/10.18129/B9.bioc.mixOmics). An optimal choice of the number of components and of the number of kept genes can be carried out using cross-validation.

First makes sure that the mixOmics Bioconductor package is installed.
```{r mixOmics}
if (!requireNamespace("limma", quietly = TRUE)){
  if (!requireNamespace("BiocManager", quietly = TRUE)){
    install.packages("BiocManager")
    }
  BiocManager::install("limma")
}
```

```{r splsda}
modsplsda=mixOmics::splsda(t(cbind(micro_S,micro_US)),groupf, ncomp = 2, 
keepX = c(100, 100))
```

Create a clustered image map (cim, i.e. heat map) to represent the results of the splsda analysis.

```{r cim, fig.width=10,fig.height=10}
mixOmics::cim(modsplsda)
```

Retrieve the names of the probesets that were selected to create the 2 splsda components

```{r selecpb}
selectedprobesets<-unique(mixOmics::selectVar(modsplsda)$name,
mixOmics::selectVar(modsplsda, comp=2)$name)
```

First makes sure that the limma Bioconductor package is installed.
```{r installlimma}
if (!requireNamespace("limma", quietly = TRUE)){
  if (!requireNamespace("BiocManager", quietly = TRUE)){
    install.packages("BiocManager")
    }
  BiocManager::install("limma")
}
```

Using the [limma](https://bioconductor.org/packages/release/bioc/html/limma.html), [doi:10.18129/B9.bioc.limma](https://doi.org/10.18129/B9.bioc.limma), plotMDS function to create the multidimensional scaling plot of distances between the probeset expression profiles that were selected using splsda.
```{r plotMDS}
limma::plotMDS(cbind(micro_S,micro_US)[selectedprobesets,])
```

### Entrez GeneIDs

The [jetset](https://cran.r-project.org/package=jetset) package enables the selection of optimal probe sets from the HG-U95Av2, HG-U133A, HG-U133 Plus 2.0, or U133 X3P microarray platforms. It requires the [org.Hs.eg.db](https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html) Bioconductor package, [doi:10.18129/B9.bioc.org.Hs.eg.db](https://doi.org/10.18129/B9.bioc.org.Hs.eg.db).


First makes sure that the jetset CRAN package and the org.Hs.eg.db Bioconductor package are installed.
```{r installpck}
if (!requireNamespace("org.Hs.eg.db", quietly = TRUE)){
  if (!requireNamespace("BiocManager", quietly = TRUE)){
      install.packages("BiocManager")
    }
  BiocManager::install("org.Hs.eg.db", version = "3.8")
}
if (!requireNamespace("jetset", quietly = TRUE)){
  install.packages("jetset")
}
```

Retrieve the optimal probesets for a given Entrez GeneID.
```{r findoptpb}
library(jetset)
resjetset=jetset::jmap("hgu133plus2", eg = sort(unique(scores.hgu133plus2$EntrezID)))
```

Select the optimal probesets from the datasets for a given Entrez GeneID.
```{r filterpb}
micro_S_jetset<-micro_S[resjetset,]
rownames(micro_S_jetset)<-names(resjetset)
micro_US_jetset<-micro_US[resjetset,]
rownames(micro_US_jetset)<-names(resjetset)
```

Then, create the 2 most discriminative components (probeset linear combinaison, i.e. scores) featuring 100 probesets each using sparse partial least squares discrimant analysis from the [mixOmics](https://www.bioconductor.org/packages/release/bioc/html/mixOmics.html) package, [doi:10.18129/B9.bioc.mixOmics](https://doi.org/10.18129/B9.bioc.mixOmics). An optimal choice of the number of components and of the number of kept genes can be carried out using cross-validation.

First makes sure that the mixOmics Bioconductor package is installed.
```{r splsdaegid}
modsplsda_jetset=mixOmics::splsda(t(cbind(micro_S_jetset,micro_US_jetset)), 
groupf, ncomp = 2, keepX = c(100, 100))
```

Create a clustered image map (cim, i.e. heat map) to represent the results of the splsda analysis.

```{r cimjetset, fig.width=10, fig.height=10}
mixOmics::cim(modsplsda_jetset)
```

Retrieve the Entrez GeneIDs that were selected to create the 2 splsda components

```{r selecegid}
selectedEntrezGeneIDs<-unique(mixOmics::selectVar(modsplsda_jetset)$name,
mixOmics::selectVar(modsplsda, comp=2)$name)
```

First makes sure that the limma Bioconductor package is installed.
```{r installlimma2}
if (!requireNamespace("limma", quietly = TRUE)){
  if (!requireNamespace("BiocManager", quietly = TRUE)){
    install.packages("BiocManager")
    }
  BiocManager::install("limma")
}
```

Using the [limma](https://bioconductor.org/packages/release/bioc/html/limma.html), [doi:10.18129/B9.bioc.limma](https://doi.org/10.18129/B9.bioc.limma), plotMDS function to create the multidimensional scaling plot of distances between the Entrez GeneID expression profiles that were selected using splsda.
```{r plotMDSegid}
limma::plotMDS(cbind(micro_S,micro_US)[selectedEntrezGeneIDs,])
```

