<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Experimental Data of Cascade Experiments in Genomics • CascadeData</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Experimental Data of Cascade Experiments in Genomics">
<meta property="og:description" content="These experimental expression data (5 leukemic CLL B-lymphocyte of aggressive form from GSE39411, &lt;doi:10.1073/pnas.1211130110&gt;), after B-cell receptor stimulation, are used as examples by packages such as the Cascade one, a modeling tool allowing gene selection, reverse engineering, and prediction in cascade networks. Jung, N., Bertrand, F., Bahram, S., Vallat, L., and Maumy-Bertrand, M. (2014) &lt;doi:10.1093/bioinformatics/btt705&gt;.">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">CascadeData</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fbertran/CascadeData/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="cascadedata-">CascadeData <img src="reference/figures/logo.png" align="right" width="200"><a class="anchor" aria-label="anchor" href="#cascadedata-"></a>
</h1></div>
</div>
<div class="section level1">
<h1 id="experimental-data-of-cascade-experiments-in-genomics">Experimental Data of Cascade Experiments in Genomics<a class="anchor" aria-label="anchor" href="#experimental-data-of-cascade-experiments-in-genomics"></a>
</h1>
<div class="section level2">
<h2 id="frédéric-bertrand-and-myriam-maumy-bertrand">Frédéric Bertrand and Myriam Maumy-Bertrand<a class="anchor" aria-label="anchor" href="#fr%C3%A9d%C3%A9ric-bertrand-and-myriam-maumy-bertrand"></a>
</h2>
<!-- badges: start -->

<p>The goal of CascadeData is to provide the experimental data <a href="https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE39411" class="external-link">GSE39411</a> in a ready to use format. Vallat L, Kemper CA, Jung N, Maumy-Bertrand M, Bertrand F, …, Bahram S, (2013), “Reverse-engineering the genetic circuitry of a cancer cell with predicted intervention in chronic lymphocytic leukemia”. <em>Proc Natl Acad Sci USA</em>, <strong>110</strong>(2):459-64, <a href="https://dx.doi.org/10.1073/pnas.1211130110" class="external-link uri">https://dx.doi.org/10.1073/pnas.1211130110</a>.</p>
<p>These are featured as examples by packages such as the Cascade one, a modeling tool allowing gene selection, reverse engineering, and prediction in cascade networks. (Jung, N., Bertrand, F., Bahram, S., Vallat, L., and Maumy-Bertrand, M., 2014, <a href="https://dx.doi.org/10.1093/bioinformatics/btt705" class="external-link uri">https://dx.doi.org/10.1093/bioinformatics/btt705</a>).</p>
<p>This website and these examples were created by F. Bertrand and M. Maumy-Bertrand.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the released version of CascadeData from <a href="https://CRAN.R-project.org" class="external-link">CRAN</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"CascadeData"</span><span class="op">)</span></span></code></pre></div>
<p>You can install the development version of CascadeData from <a href="https://github.com" class="external-link">github</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"fbertran/CascadeData"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<div class="section level3">
<h3 id="probesets">Probesets<a class="anchor" aria-label="anchor" href="#probesets"></a>
</h3>
<p>Two data frames with 54613 probesets with repeated measurements on 6 independent subjects at 4 time points.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">micro_S</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">micro_S</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'data.frame':    54613 obs. of  24 variables:</span></span>
<span><span class="co">#&gt;  $ N1_S_T60 : num  136.1 32 78 201.8 16.3 ...</span></span>
<span><span class="co">#&gt;  $ N1_S_T90 : num  116.6 43.3 63.5 209.2 8 ...</span></span>
<span><span class="co">#&gt;  $ N1_S_T210: num  127.6 31.3 57.9 208.8 15.8 ...</span></span>
<span><span class="co">#&gt;  $ N1_S_T390: num  126.8 43.8 73.2 228.2 15.8 ...</span></span>
<span><span class="co">#&gt;  $ N2_S_T60 : num  142.2 25.4 53.2 165.3 17.8 ...</span></span>
<span><span class="co">#&gt;  $ N2_S_T90 : num  132.5 35.4 97.1 222.3 10.7 ...</span></span>
<span><span class="co">#&gt;  $ N2_S_T210: num  122.8 33.3 66.5 175.4 12.9 ...</span></span>
<span><span class="co">#&gt;  $ N2_S_T390: num  132.1 55.7 69.7 178.5 13.9 ...</span></span>
<span><span class="co">#&gt;  $ N3_S_T60 : num  134 60 52.1 214.6 6.6 ...</span></span>
<span><span class="co">#&gt;  $ N3_S_T90 : num  157.1 48.7 59.9 279.5 3.7 ...</span></span>
<span><span class="co">#&gt;  $ N3_S_T210: num  49.1 45.7 44.1 181.2 1.3 ...</span></span>
<span><span class="co">#&gt;  $ N3_S_T390: num  139.8 41.4 51.8 264.9 19.5 ...</span></span>
<span><span class="co">#&gt;  $ N4_S_T60 : num  157.4 66.8 74 214.1 26.1 ...</span></span>
<span><span class="co">#&gt;  $ N4_S_T90 : num  158.3 44.8 56.8 222.7 17.9 ...</span></span>
<span><span class="co">#&gt;  $ N4_S_T210: num  145.9 42.5 59.8 221.8 18.5 ...</span></span>
<span><span class="co">#&gt;  $ N4_S_T390: num  171.2 59.2 62.3 193.3 25.1 ...</span></span>
<span><span class="co">#&gt;  $ N5_S_T60 : num  130.1 60.8 63.8 208.5 18.6 ...</span></span>
<span><span class="co">#&gt;  $ N5_S_T90 : num  119.7 41.8 47.2 222.8 26.9 ...</span></span>
<span><span class="co">#&gt;  $ N5_S_T210: num  121.3 40.5 62.7 226 21.5 ...</span></span>
<span><span class="co">#&gt;  $ N5_S_T390: num  110 44 63 227.8 27.2 ...</span></span>
<span><span class="co">#&gt;  $ N6_S_T60 : num  114.5 60.7 79.5 219.4 22.1 ...</span></span>
<span><span class="co">#&gt;  $ N6_S_T90 : num  144 38.9 79.4 234.8 28.4 ...</span></span>
<span><span class="co">#&gt;  $ N6_S_T210: num  128.4 38.7 63.4 241.9 30.5 ...</span></span>
<span><span class="co">#&gt;  $ N6_S_T390: num  132.7 52.3 66.6 218.7 19.2 ...</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">micro_US</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">micro_US</span><span class="op">)</span></span>
<span><span class="co">#&gt; 'data.frame':    54613 obs. of  24 variables:</span></span>
<span><span class="co">#&gt;  $ N1_US_T60 : num  103.2 26 70.7 213.7 13.7 ...</span></span>
<span><span class="co">#&gt;  $ N1_US_T90 : num  133.7 34.9 71.2 168.9 17.2 ...</span></span>
<span><span class="co">#&gt;  $ N1_US_T210: num  157.3 44.2 59.4 175.1 27.8 ...</span></span>
<span><span class="co">#&gt;  $ N1_US_T390: num  179.4 44.1 62.7 225.1 10.5 ...</span></span>
<span><span class="co">#&gt;  $ N2_US_T60 : num  120.7 47.4 85.8 156.7 16.7 ...</span></span>
<span><span class="co">#&gt;  $ N2_US_T90 : num  150.5 40.4 74.4 200.4 25.8 ...</span></span>
<span><span class="co">#&gt;  $ N2_US_T210: num  153.5 51.8 71.7 181.6 23 ...</span></span>
<span><span class="co">#&gt;  $ N2_US_T390: num  174.8 61.6 66.8 224.5 12.2 ...</span></span>
<span><span class="co">#&gt;  $ N3_US_T60 : num  145 64.5 50.8 184.5 17.5 ...</span></span>
<span><span class="co">#&gt;  $ N3_US_T90 : num  162.2 58.7 51.9 219.1 21.5 ...</span></span>
<span><span class="co">#&gt;  $ N3_US_T210: num  177.8 35.9 58.3 250 12.7 ...</span></span>
<span><span class="co">#&gt;  $ N3_US_T390: num  189.6 50.5 62.8 262 20.1 ...</span></span>
<span><span class="co">#&gt;  $ N4_US_T60 : num  136 60.9 67.8 227.1 29.6 ...</span></span>
<span><span class="co">#&gt;  $ N4_US_T90 : num  135.2 62.2 65.3 221.1 18.2 ...</span></span>
<span><span class="co">#&gt;  $ N4_US_T210: num  165.3 62.7 63.3 212.1 21.4 ...</span></span>
<span><span class="co">#&gt;  $ N4_US_T390: num  154.3 47.1 68.9 199.7 9.3 ...</span></span>
<span><span class="co">#&gt;  $ N5_US_T60 : num  130.9 57.5 56.5 199 20.5 ...</span></span>
<span><span class="co">#&gt;  $ N5_US_T90 : num  122.3 49.4 62 226.8 15.4 ...</span></span>
<span><span class="co">#&gt;  $ N5_US_T210: num  184.8 57.2 53.4 191.2 18.1 ...</span></span>
<span><span class="co">#&gt;  $ N5_US_T390: num  174.6 68 69.1 214 16.5 ...</span></span>
<span><span class="co">#&gt;  $ N6_US_T60 : num  142.8 67.9 73.3 220.1 24.7 ...</span></span>
<span><span class="co">#&gt;  $ N6_US_T90 : num  148.1 46.4 79.9 257.7 20.7 ...</span></span>
<span><span class="co">#&gt;  $ N6_US_T210: num  198.1 53.9 68.6 223.3 18.9 ...</span></span>
<span><span class="co">#&gt;  $ N6_US_T390: num  155.8 42.6 63.2 236.7 15.2 ...</span></span></code></pre></div>
<p>Some preliminary between group comparison. First create grouping factor.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">groupf</span><span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"S"</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">micro_S</span><span class="op">)</span><span class="op">)</span>,<span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"US"</span>,<span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">micro_US</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Then, create the 2 most discriminative components (probeset linear combinaison, i.e. scores) featuring 100 probesets each using sparse partial least squares discrimant analysis from the <a href="https://www.bioconductor.org/packages/release/bioc/html/mixOmics.html" class="external-link">mixOmics</a> package, <a href="https://doi.org/10.18129/B9.bioc.mixOmics" class="external-link">doi:10.18129/B9.bioc.mixOmics</a>. An optimal choice of the number of components and of the number of kept genes can be carried out using cross-validation.</p>
<p>First makes sure that the mixOmics Bioconductor package is installed.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"limma"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"limma"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modsplsda</span><span class="op">=</span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">micro_S</span>,<span class="va">micro_US</span><span class="op">)</span><span class="op">)</span>,<span class="va">groupf</span>, ncomp <span class="op">=</span> <span class="fl">2</span>, </span>
<span>keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Create a clustered image map (cim, i.e. heat map) to represent the results of the splsda analysis.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/cim.html" class="external-link">cim</a></span><span class="op">(</span><span class="va">modsplsda</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-cim-1.png" title="plot of chunk cim" alt="plot of chunk cim" width="100%"></p>
<p>Retrieve the names of the probesets that were selected to create the 2 splsda components</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selectedprobesets</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">modsplsda</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,</span>
<span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">modsplsda</span>, comp<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span></code></pre></div>
<p>First makes sure that the limma Bioconductor package is installed.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"limma"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"limma"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Using the <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" class="external-link">limma</a>, <a href="https://doi.org/10.18129/B9.bioc.limma" class="external-link">doi:10.18129/B9.bioc.limma</a>, plotMDS function to create the multidimensional scaling plot of distances between the probeset expression profiles that were selected using splsda.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">micro_S</span>,<span class="va">micro_US</span><span class="op">)</span><span class="op">[</span><span class="va">selectedprobesets</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plotMDS-1.png" title="plot of chunk plotMDS" alt="plot of chunk plotMDS" width="100%"></p>
</div>
<div class="section level3">
<h3 id="entrez-geneids">Entrez GeneIDs<a class="anchor" aria-label="anchor" href="#entrez-geneids"></a>
</h3>
<p>The <a href="https://cran.r-project.org/package=jetset" class="external-link">jetset</a> package enables the selection of optimal probe sets from the HG-U95Av2, HG-U133A, HG-U133 Plus 2.0, or U133 X3P microarray platforms. It requires the <a href="https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html" class="external-link">org.Hs.eg.db</a> Bioconductor package, <a href="https://doi.org/10.18129/B9.bioc.org.Hs.eg.db" class="external-link">doi:10.18129/B9.bioc.org.Hs.eg.db</a>.</p>
<p>First makes sure that the jetset CRAN package and the org.Hs.eg.db Bioconductor package are installed.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>      <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span>, version <span class="op">=</span> <span class="st">"3.8"</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"jetset"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"jetset"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Retrieve the optimal probesets for a given Entrez GeneID.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="http://www.cbs.dtu.dk/biotools/jetset/" class="external-link">jetset</a></span><span class="op">)</span></span>
<span><span class="va">resjetset</span><span class="op">=</span><span class="fu">jetset</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/jetset/man/jmap.html" class="external-link">jmap</a></span><span class="op">(</span><span class="st">"hgu133plus2"</span>, eg <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">scores.hgu133plus2</span><span class="op">$</span><span class="va">EntrezID</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Select the optimal probesets from the datasets for a given Entrez GeneID.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">micro_S_jetset</span><span class="op">&lt;-</span><span class="va">micro_S</span><span class="op">[</span><span class="va">resjetset</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">micro_S_jetset</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">resjetset</span><span class="op">)</span></span>
<span><span class="va">micro_US_jetset</span><span class="op">&lt;-</span><span class="va">micro_US</span><span class="op">[</span><span class="va">resjetset</span>,<span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">micro_US_jetset</span><span class="op">)</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">resjetset</span><span class="op">)</span></span></code></pre></div>
<p>Then, create the 2 most discriminative components (probeset linear combinaison, i.e. scores) featuring 100 probesets each using sparse partial least squares discrimant analysis from the <a href="https://www.bioconductor.org/packages/release/bioc/html/mixOmics.html" class="external-link">mixOmics</a> package, <a href="https://doi.org/10.18129/B9.bioc.mixOmics" class="external-link">doi:10.18129/B9.bioc.mixOmics</a>. An optimal choice of the number of components and of the number of kept genes can be carried out using cross-validation.</p>
<p>First makes sure that the mixOmics Bioconductor package is installed.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">modsplsda_jetset</span><span class="op">=</span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/splsda.html" class="external-link">splsda</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/t.html" class="external-link">t</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">micro_S_jetset</span>,<span class="va">micro_US_jetset</span><span class="op">)</span><span class="op">)</span>, </span>
<span><span class="va">groupf</span>, ncomp <span class="op">=</span> <span class="fl">2</span>, keepX <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">100</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Create a clustered image map (cim, i.e. heat map) to represent the results of the splsda analysis.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/cim.html" class="external-link">cim</a></span><span class="op">(</span><span class="va">modsplsda_jetset</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-cimjetset-1.png" title="plot of chunk cimjetset" alt="plot of chunk cimjetset" width="100%"></p>
<p>Retrieve the Entrez GeneIDs that were selected to create the 2 splsda components</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">selectedEntrezGeneIDs</span><span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">modsplsda_jetset</span><span class="op">)</span><span class="op">$</span><span class="va">name</span>,</span>
<span><span class="fu">mixOmics</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/mixOmics/man/selectVar.html" class="external-link">selectVar</a></span><span class="op">(</span><span class="va">modsplsda</span>, comp<span class="op">=</span><span class="fl">2</span><span class="op">)</span><span class="op">$</span><span class="va">name</span><span class="op">)</span></span></code></pre></div>
<p>First makes sure that the limma Bioconductor package is installed.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"limma"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span>    <span class="op">}</span></span>
<span>  <span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"limma"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<p>Using the <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" class="external-link">limma</a>, <a href="https://doi.org/10.18129/B9.bioc.limma" class="external-link">doi:10.18129/B9.bioc.limma</a>, plotMDS function to create the multidimensional scaling plot of distances between the Entrez GeneID expression profiles that were selected using splsda.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">limma</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/limma/man/plotMDS.html" class="external-link">plotMDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="va">micro_S</span>,<span class="va">micro_US</span><span class="op">)</span><span class="op">[</span><span class="va">selectedEntrezGeneIDs</span>,<span class="op">]</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-plotMDSegid-1.png" title="plot of chunk plotMDSegid" alt="plot of chunk plotMDSegid" width="100%"></p>
</div>
</div>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=CascadeData" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/fbertran/CascadeData/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/fbertran/CascadeData/issues/" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>GPL (&gt;= 2)</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CascadeData</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Frederic Bertrand <br><small class="roles"> Maintainer, author </small> <a href="https://orcid.org/0000-0002-0837-8281" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Myriam Maumy-Bertrand <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4615-1512" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li><a href="authors.html">More about authors...</a></li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://lifecycle.r-lib.org/articles/stages.html" class="external-link"><img src="https://img.shields.io/badge/lifecycle-stable-green.svg" alt="Lifecycle: stable"></a></li>
<li><a href="https://www.repostatus.org/#active" class="external-link"><img src="https://www.repostatus.org/badges/latest/active.svg" alt="Project Status: Active – The project has reached a stable, usable state and is being actively developed."></a></li>
<li><a href="https://github.com/fbertran/CascadeData/actions" class="external-link"><img src="https://github.com/fbertran/CascadeData/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/fbertran/CascadeData?branch=master" class="external-link"><img src="https://codecov.io/gh/fbertran/CascadeData/branch/master/graph/badge.svg" alt="Codecov test coverage"></a></li>
<li><a href="https://cran.r-project.org/package=CascadeData" class="external-link"><img src="https://www.r-pkg.org/badges/version/CascadeData" alt="CRAN status"></a></li>
<li><a href="https://cran.r-project.org/package=CascadeData" class="external-link"><img src="https://cranlogs.r-pkg.org/badges/CascadeData" alt="CRAN RStudio mirror downloads"></a></li>
<li><a href="https://github.com/fbertran/CascadeData" class="external-link"><img src="https://img.shields.io/github/stars/fbertran/CascadeData?style=social" alt="GitHub Repo stars"></a></li>
<li><a href="https://zenodo.org/badge/latestdoi/167803258" class="external-link"><img src="https://zenodo.org/badge/167803258.svg" alt="DOI"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Frederic Bertrand, Myriam Maumy-Bertrand.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
